### PR TITLE
Fix/meshopt

### PR DESCRIPTION
**Pull request by yiwenxue**
Re: #https://github.com/cocos/3d-tasks/issues/17619

### Changelog

* add missing part.
* add fallback for platform doesn't support half as vertex attribute
* complete document

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

**Review by stanleyljl — APPROVED** [review]

**Comment by yiwenxue** [comment]
@cocos-robot run test cases

**Comment by yiwenxue** [comment]
@cocos-robot run test cases

**Comment by sushanhong** [comment]
@cocos-robot run test cases

**Comment by sushanhong** [comment]
@cocos-robot run test cases

File: native/cocos/3d/assets/Mesh.cpp [hunk 1]
```diff
@@ -247,6 +251,140 @@ void convertRG32FToRG16F(const float *src, uint16_t *dst) {
...
+    };
+
+    for (auto &bundle : structInfo.vertexBundles) {
+        auto &view = bundle.view;
+        auto &attrs = bundle.attributes;
+        auto oldAttrs = attrs;
+        std::vector<uint32_t> strides;
+        std::vector<bool> dequantizes;
+        std::vector<DataReaderCallback> readers;
+        for (uint32_t i = 0; i < attrs.size(); ++i) {
```
Comment (dumganhar) [inline native/cocos/3d/assets/Mesh.cpp:305]: Add reserve for these vectors?
```
strides.reserve(attrs.size());
dequantizes.reserve(attrs.size());
readers.reserve(attrs.size());
```

**Comment by sushanhong** [comment]
@cocos-robot run test cases

**Comment by sushanhong** [comment]
@cocos-robot run test cases

**Comment by sushanhong** [comment]
@cocos-robot run test cases

File: native/cocos/3d/assets/Mesh.cpp (hunk 1 above)
Comment (yiwenxue) [reply native/cocos/3d/assets/Mesh.cpp:305]: resolved

**Pull request merged**